### PR TITLE
api: add instances?tail=true to get latest WFIs (alternative to #591)

### PR DESCRIPTION
**Pull request by honnix**
# Hey, I just made a Pull Request!

## Description

Allow fetching the latest (by parameter) instances of a workflow.

Alternative to #591.

## Motivation and Context

Users want an easy way to to get the latest instances of a workflow for health checking purposes.

## Checklist for PR author(s)

- [x] Changes are covered by unit test
- [ ] All tests pass
- [ ] Code coverage check passes
- [ ] Error handling is tested
- [ ] Errors are handled at the appropriate layer
- [ ] Errors that cannot be handled where they occur are propagated
- [ ] (optional) Changes are covered by system test
- [ ] Relevant documentation updated
- [ ] This PR has NO breaking change to public API
- [ ] This PR has breaking change to public API and it is documented

## Checklist for PR reviewer(s)

- [ ] This PR has been incorporated in release note for the coming version
- [ ] Risky changes introduced by this PR have been all considered

### CODE REVIEW

File: styx-common/src/main/java/com/spotify/styx/storage/BigtableStorage.java [hunk 1]
```diff
@@ -116,7 +116,7 @@ void writeEvent(SequenceEvent sequenceEvent) throws IOException {
 
       if (!Strings.isNullOrEmpty(offset)) {
         final WorkflowInstance offsetInstance = WorkflowInstance.create(workflowId, offset);
-        scan.setStartRow(Bytes.toBytes(offsetInstance.toKey() + '#'));
+        scan.withStartRow(Bytes.toBytes(offsetInstance.toKey() + '#'), true);
```
Comment (honnix) [inline styx-common/src/main/java/com/spotify/styx/storage/BigtableStorage.java]: `setXxxRow` are deprecated.

**Comment by honnix** [comment]
@danielnorberg PTAL

File: styx-common/src/main/java/com/spotify/styx/storage/BigtableStorage.java (hunk 1 above)
Comment (honnix) [reply styx-common/src/main/java/com/spotify/styx/storage/BigtableStorage.java]: Reverted. Better to be a separated PR.

**Comment by honnix** [comment]
Not sure what happened to codecov...

File: styx-common/src/main/java/com/spotify/styx/util/TimeUtil.java [hunk 2]
```diff
@@ -174,6 +175,36 @@ public static Instant nextInstant(Instant instant, Schedule schedule) {
...
+   */
+  public static Instant instantWithOffsetTo(Instant referenceInstant, int offset, Schedule schedule) {
+    if (!isAligned(referenceInstant, schedule)) {
+      throw new IllegalArgumentException("unaligned reference instant");
+    }
+
+    final ExecutionTime executionTime = ExecutionTime.forCron(cron(schedule));
+
+    Instant currentInstant = referenceInstant;
+    for (int i = 0; i < Math.abs(offset); i++) {
```
Comment (danielnorberg) [inline styx-common/src/main/java/com/spotify/styx/util/TimeUtil.java]: Offsetting the instant can be done more efficiently: https://github.com/spotify/styx/pull/591/files#diff-83fdfa5c37381e562bd0705651c031a6R271

**Comment by honnix** [comment]
Let's continue with #591

**Pull request closed without merging**